Backend/src/service/facebook.ts

```diff
@@ -1,36 +1,85 @@
 const facebookConfig = require('../config/facebook')
 const axiosModule = require('axios')
 
+import { error } from 'console'
+import { database, userCollectionName } from '../database/mongo'
 
+import { User } from '../dto/user'
 
-let userTokens = ""
+let userToken = ""
 
-function auth(request : {},response : any){
-    return response.redirect('https://www.facebook.com/v15.0/dialog/oauth?response_type=code&client_id=575427578042258&redirect_uri=https://57ef-2401-4900-1cb9-a1d9-8c5e-2e5c-b745-5c45.ngrok-free.app/auth/callback')
+const collection = database.collection(userCollectionName);
+
+function auth(request: {}, response: any) {
+  return response.redirect('https://www.facebook.com/v15.0/dialog/oauth?response_type=code&client_id=' + facebookConfig.client_id + '&redirect_uri=' + facebookConfig.redirect_uri)
 }
 
-function authCallback(request : any,response : any){
-    const code = request.query.code;
+async function authCallback(request: any, response: any) {
+  const code = request.query.code;
```

Comment:
it may break if request doesn't contain query. use ?? operator or use strict request type